app/src/main/java/org/dhis2/data/forms/dataentry/fields/spinner/SpinnerHolder.java

```diff
@@ -57,6 +60,19 @@ public void update(SpinnerViewModel viewModel) {
         binding.optionSetView.setOnClickListener(this);
         label = new StringBuilder().append(viewModel.label());
         initFieldFocus();
+
+        if(!isSearchMode){
+            assignBackgroundColorByLegend();
+        }
+    }
+
+    private void assignBackgroundColorByLegend() {
+        if (viewModel.colorByLegend() != null && viewModel.colorByLegend() != ""){
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.